### ISSUE TITLE
 Improve the recognition efficiency

### ISSUE THREAD

**Issue by njustzjw** [ShuangLI59/person_search#41]
I got about 0.8 seconds  for an image, so what can I do to improve the recognition efficiency, except for the increasing in the number of GPU?

**Comment by Cysu** [comment]
One way is to increase the GPU utilization, by for example, batching multiple images together. Current code supports only one image per mini-batch. If we can rewrite this to support multiple images, I think it could significantly improve the efficiency.

By the way, you may inspect your GPU utilization by `nvidia-smi` when running the script.

**Comment by njustzjw** [comment]
In my test, I found that the resolution ratio of the image does not affect the efficiency of recognition, so you mean to take multiple images to do split joint  into a new image, and then do the identification?

**Comment by Cysu** [comment]
No. What I mean is that normally CNN accepts input shape like NxCxHxW, where N is the number of images in a mini-batch. However, current code only [supports](https://github.com/ShuangLI59/person_search/blob/master/lib/roi_data_layer/minibatch.py#L33-L35) N = 1. We need to rewrite and extend the library to make it support N > 1. Possibly we need to handle different image sizes in a mini-batch, by for example, properly padding and cropping them.

**Comment by njustzjw** [comment]
Thank you very much! I have found that people with similar colors are prone to become false identification. Is this normal?Is it related to that the database is too large?

**Comment by Cysu** [comment]
Yeah, it's possible. Colors would be quite important evidence. Larger gallery size could also degrade the performance.

**Comment by njustzjw** [comment]
Yes,I got it. Maybe I need to train my data alone, because I added my data on your data before.
In train process, how much does the value of loss should drop to ?Then the performance will be good.

**Comment by Cysu** [comment]
I would say it's hard to tell the performance only by inspecting the training loss. It's better to have a validation set, on which you can evaluate the mAP or CMC during the training process, for example, every 10000 iterations.
